### PR TITLE
Adding a resource and doc for aws_sts_caller_identity

### PR DESCRIPTION
**Pull request by tmonk42**
I have found this useful in changing behavior when scannign GovCloud accounts

### CODE REVIEW

File: docs/resources/aws_sts_caller_identify.md.erb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      it { should be_govcloud }
+    end
+
+<br>
+
+## Examples
+
+The following examples show how to use this InSpec audit resource.
+
+    # Skip a check if GovCloud
```
Comment (jerryaldrichiii) [inline docs/resources/aws_sts_caller_identify.md.erb:31]: I would rewrite this as:

```
# Skipping a test if on GovCloud
control 'aws_sts_caller_identity_example' do
  impact 1.0
  title 'Check root MFA token'

  only_if('Skip Root User MFA check on GovCloud') do
    !aws_sts_caller_identity.govcloud
  end

  describe aws_iam_root_user do
    it { should have_mfa_enabled }  
  end
end
```

**Review by jerryaldrichiii — APPROVED** [review]
Looks good to me! Just one suggestion for an example change.

**Comment by clintoncwolfe** [comment]
Looks like the linter doesn't like it - a quick `bundle exec rubocop -a` and a commit/push should fix that.

**Comment by ghost** [comment]
Thanks for your contribution! The @inspec/inspec-cloud-devs team have recently revisited AWS and created a separate resource pack that lives here - https://github.com/inspec/inspec-aws. As part of this we reviewed open PRs and ensured consistent coverage in the new AWS resource pack. Please take a look at the related PR here: https://github.com/inspec/inspec-aws/pull/11)

**Pull request closed without merging**